### DISCUSSION TITLE
Static IP in cloudstack instance

### DISCUSSION THREAD

**Discussion by j3cky** [apache/cloudstack#9342] [Q&A] [answered]
Hi,
i would like to know if cloudstack support configure static ip in instance.?
i cannot find how to configure static ip,only dhcp.
i would like to create network offering without dhcp service.how can cloudstack configure static ip inside the VM?

**Comment by pcfriek1987** [comment]
I think that would be up to setting up a VLAN then to your outside network, the VRR only does DHCP.

**Comment by weizhouapache** [comment]
@j3cky 
what did you mean `static` ? do you want to assign a specific IP to the vm ?

**Comment by j3cky** [comment]
@weizhouapache 
yes, after instance deployed example centos 7, we get config in /etc/sysconfig/network-scripts/ifcfg-eth0 using dhcp.

BOOTPROTO=dhcp
DEVICE=eth0
HWADDR=02:02:01:66:00:27
ONBOOT=yes
STARTMODE=auto
TYPE=Ethernet
USERCTL=no

is it possible to have it as static ip like this

BOOTPROTO=static
IPADDR=192.168.2.203 #depend on assigned ip when we deploy.
PREFIX=24
DEVICE=eth0
HWADDR=02:02:01:66:00:27
ONBOOT=yes
STARTMODE=auto
TYPE=Ethernet
USERCTL=no

**Reply by weizhouapache** [reply]
You can make the changes you mentioned, then stop the vm and create a template from its volume.
But all vms created from the template will have the same IP. What's your use case?

**Comment by j3cky** [comment]
so, we are service provider and if we enable dhcp service then vr wil be created and it will consume compute resources.so we would like to use configdrive but seem it is not support configure network interface to static ip.

**Reply by weizhouapache** [reply, marked as answer]
> so, we are service provider and if we enable dhcp service then vr wil be created and it will consume compute resources.so we would like to use configdrive but seem it is not support configure network interface to static ip.

@j3cky 
there is a work in progress which might be what you want
https://github.com/apache/cloudstack/pull/9329

**Reply by j3cky** [reply]
@weizhouapache thanks for your update.

**Comment by returntrip** [comment]
@j3cky  Have you been able to actually set a static ip using a network offering and configdrive?

**Comment by j3cky** [comment]
@returntrip 
no, config drive for ip address still need dhcp service.

**Reply by weizhouapache** [reply]
> @returntrip no, config drive for ip address still need dhcp service.

yes, Configdrive as dhcp and dns provider
